inference-engine/src/mkldnn_plugin/nodes/mkldnn_shuffle_channels_node.cpp

```diff
@@ -0,0 +1,712 @@
+            else if (axis == 2) shuffle_mode = ShuffleMode::ShuffleDepth;
+            else if (axis == 3) shuffle_mode = ShuffleMode::ShuffleHeight;
+            else if (axis == 4) shuffle_mode = ShuffleMode::ShuffleWidth;
+        } else if (dims_size == 4) {
+            SET_DIM_VALUE(dst_dims[0], dst_dims[1], 1, dst_dims[2], dst_dims[3]);
+            if (axis == 0) shuffle_mode = ShuffleMode::ShuffleBatch;
+            else if (axis == 1) shuffle_mode = ShuffleMode::ShuffleChannel;
+            else if (axis == 2) shuffle_mode = ShuffleMode::ShuffleHeight;
+            else if (axis == 3) shuffle_mode = ShuffleMode::ShuffleWidth;
+        } else if (dims_size == 3) {
+            SET_DIM_VALUE(1, dst_dims[0], 1, dst_dims[1], dst_dims[2]);
+            if (axis == 0) shuffle_mode = ShuffleMode::ShuffleChannel;
+            else if (axis == 1) shuffle_mode = ShuffleMode::ShuffleHeight;
+            else if (axis == 2) shuffle_mode = ShuffleMode::ShuffleWidth;
+        } else if (dims_size == 2) {
+            SET_DIM_VALUE(dst_dims[0], dst_dims[1], 1, 1, 1);
+            if (axis == 0) shuffle_mode = ShuffleMode::ShuffleHeight;
+            else if (axis == 1) shuffle_mode = ShuffleMode::ShuffleWidth;
+        } else {
+            SET_DIM_VALUE(1, dst_dims[0], 1, 1, 1);
+            if (axis == 0) shuffle_mode = ShuffleMode::ShuffleChannel;
+        }
+    }
+}
+
+void MKLDNNShuffleChannelsNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    Precision inputPrecision = getCnnLayer()->insData[0].lock()->getPrecision();
+    Precision outputPrecision = getCnnLayer()->outData[0]->getPrecision();
+
+    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(inputPrecision);
+    auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(outputPrecision);
+
+    src_data_size = MKLDNNExtensionUtils::sizeOfDataType(inputDataType);
+    dst_data_size = MKLDNNExtensionUtils::sizeOfDataType(outputDataType);
+
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = false;
+    config.inConfs.resize(1);
+    config.outConfs.resize(1);
+    config.inConfs[0].constant = false;
+    config.outConfs[0].constant = false;
+    config.inConfs[0].inPlace = -1;
+    config.outConfs[0].inPlace = -1;
+
+    auto pushDesc = [&](memory::format inFormat, memory::format outFormat, memory::data_type inDataType, memory::data_type outDataType) {
+        config.inConfs[0].desc = MKLDNNMemoryDesc(getParentEdgeAt(0)->getDims(), inDataType, inFormat);
+        config.outConfs[0].desc = MKLDNNMemoryDesc(getChildEdgeAt(0)->getDims(), outDataType, outFormat);
+        supportedPrimitiveDescriptors.push_back({config, impl_desc_type::unknown, outFormat});
+    };
+
+    // vgatherdps in kernel neither support data type u8 nor s8, and it doesn't support isa sse42.
+    bool kernel_gather = shuffle_mode == ShuffleMode::ShuffleWidth || shuffle_mode == ShuffleMode::ShuffleChannel;
```

Comment:
Fixed. kernel_gather was a flag to decide whether instructions of gather can be used in JIT kernel, because instructions of gather are not fully supported by all data type and isa. Anyway, it is no longer actual, since imitated gather ops is added where it is not supported. Gather can alway be done in JIT kernel when shuffling innermost dims now. Thanks Dmitry!